src/workerd/api/node/path-test.js

```diff
@@ -0,0 +1,543 @@
+          const key = actualKeys[i];
+          if (!expectedKeys.includes(key) || actual[key] !== expected[key]) {
+            failed = true;
+            break;
+          }
+        }
+      }
+      if (failed)
+        failures.push(`\n${message}`);
+    });
+    strictEqual(failures.length, 0, failures.join(''));
+
+    function checkErrors() {
+      errors.forEach(({ method, input }) => {
+        throws(() => {
+          path[method].apply(path, input);
+        }, {
+          code: 'ERR_INVALID_ARG_TYPE',
+          name: 'TypeError'
+        });
+      });
+    }
+
+    function checkParseFormat(paths) {
+      paths.forEach(([element, root]) => {
+        const output = path.parse(element);
+        strictEqual(typeof output.root, 'string');
+        strictEqual(typeof output.dir, 'string');
+        strictEqual(typeof output.base, 'string');
+        strictEqual(typeof output.ext, 'string');
+        strictEqual(typeof output.name, 'string');
+        strictEqual(path.format(output), element);
+        strictEqual(output.root, root);
+        ok(output.dir.startsWith(output.root));
+        strictEqual(output.dir, output.dir ? path.dirname(element) : '');
+        strictEqual(output.base, path.basename(element));
+        strictEqual(output.ext, path.extname(element));
+      });
+    }
+
+    function checkFormat(testCases) {
+      testCases.forEach(([element, expect]) => {
+        strictEqual(path.format(element), expect);
+      });
+
+      [null, undefined, 1, true, false, 'string'].forEach((pathObject) => {
+        throws(() => {
+          path.format(pathObject);
+        }, {
+          code: 'ERR_INVALID_ARG_TYPE',
+          name: 'TypeError',
+        });
+      });
+    }
+
+    // See https://github.com/nodejs/node/issues/44343
+    strictEqual(path.format({ name: 'x', ext: 'png' }), 'x.png');
+    strictEqual(path.format({ name: 'x', ext: '.png' }), 'x.png');
+  }
+};
+
+export const test_path_normalize = {
+  test(ctrl, env, ctx) {
+    strictEqual(path.posix.normalize('./fixtures///b/../b/c.js'), 'fixtures/b/c.js');
+    strictEqual(path.posix.normalize('/foo/../../../bar'), '/bar');
+    strictEqual(path.posix.normalize('a//b//../b'), 'a/b');
+    strictEqual(path.posix.normalize('a//b//./c'), 'a/b/c');
+    strictEqual(path.posix.normalize('a//b//.'), 'a/b');
+    strictEqual(path.posix.normalize('/a/b/c/../../../x/y/z'), '/x/y/z');
+    strictEqual(path.posix.normalize('///..//./foo/.//bar'), '/foo/bar');
+    strictEqual(path.posix.normalize('bar/foo../../'), 'bar/');
+    strictEqual(path.posix.normalize('bar/foo../..'), 'bar');
+    strictEqual(path.posix.normalize('bar/foo../../baz'), 'bar/baz');
+    strictEqual(path.posix.normalize('bar/foo../'), 'bar/foo../');
+    strictEqual(path.posix.normalize('bar/foo..'), 'bar/foo..');
+    strictEqual(path.posix.normalize('../foo../../../bar'), '../../bar');
+    strictEqual(path.posix.normalize('../.../.././.../../../bar'),
```

Comment:
The '...' draws attention to the reader, but is expected to be just a plain path component under POSIX. Some test cases might that leave '...' in the result might make that clearer (or a comment).

```suggestion
    strictEqual(path.posix.normalize('bar/.../foo'), 'bar/.../foo');
    strictEqual(path.posix.normalize('bar/foo/...'), 'bar/foo/...');
    strictEqual(path.posix.normalize('../.../.././.../../../bar'),
```

Reply:
These tests are pulled directly from Node.js and I really don't want to diverge too much.